world-explore/src/app/home/home.page.ts

```diff
@@ -75,10 +66,16 @@ export class HomePage implements OnInit {
   }
 
   filterCountries() {
-    this.filteredCountries = this.countries.filter(country => 
-      country.country.toLowerCase().includes(this.searchTerm.toLowerCase()) &&
-      (!this.showFavorites || this.favoriteCountries.has(country.country))
-    );
+    if (this.showFavorites) {
```

Comment:
1- Eliminates the need for if and else if by using a single filtering expression.
2- Reduces redundant iterations over the countries array by performing filtering in a single pass.